docs/configuration/http_conn_man/route_config/route.rst

```diff
@@ -117,6 +118,24 @@ case_sensitive
   *(optional, boolean)* Indicates that prefix/path matching should be case sensitive. The default
   is true.
 
+.. _config_http_conn_man_route_table_route_allow_websocket_upgrade:
+
+allow_websocket_upgrade
+  *(optional, boolean)* Indicates that a HTTP/1.1 client connection to this particular route
+  should be allowed to upgrade to a WebSocket connection. The default is false.
+
+  .. attention::
+
+    If set to true, Envoy will expect the first request matching this route to contain WebSocket
+    upgrade headers. If the headers are not present, the connection will be rejected. If set to
+    true, Envoy will setup plain TCP proxying between the client and the upstream server. Hence,
+    an upstream server that rejects the WebSocket upgrade request is also responsible for closing
+    the associated connection. Until then, Envoy will continue to proxy data from the client to
+    the upstream server.
```

Comment:
If the upstream closes its connection after rejecting the upgrade, Envoy will flush the pipe and close downstream as well (standard tcp proxy semantics). Until upstream keeps the connection open, Envoy does standard tcp proxying - buffer bloat is subject to flow control settings.